manifests/user.pp

```diff
@@ -205,139 +203,131 @@
 ) {
 
   assert_type(Accounts::User::Name, $name)
-  if $home {
-    $_home = $home
-  } elsif $name == 'root' {
-    $_home = $::osfamily ? {
-      'Solaris' => '/',
-      default   => '/root',
-    }
-  } else {
-    $_home = $::osfamily ? {
-      'Solaris' => "/export/home/${name}",
-      default   => "/home/${name}",
-    }
+
+  $_home = $home ? {
+    undef => $name ? {
+      'root' => $facts['os']['family'] ? {
+        'Solaris' => '/',
+        default   => '/root',
+      },
+      default => $facts['os']['family'] ? {
+        'Solaris' => "/export/home/${name}",
+        default   => "/home/${name}",
+      },
+    },
+    default => $home,
   }
 
-  if $locked {
-    case $::operatingsystem {
-      'debian', 'ubuntu' : {
-        $_shell = '/usr/sbin/nologin'
-      }
-      'solaris' : {
-        $_shell = '/usr/bin/false'
+  if $ensure == 'absent' {
+    user { $name:
+      ensure     => 'absent',
+      forcelocal => $forcelocal,
+      home       => $_home,
+      managehome => $purge_user_home,
+    }
+    User[$name] -> Group <| |>
+    if $purge_user_home {
+      accounts::home_dir { $_home:
+        ensure => 'absent',
+        group  => $group,
+        user   => $name,
+
       }
-      default : {
-        $_shell = '/sbin/nologin'
+    }
+    if $create_group {
+      # Only remove the group if it is the same as user name as it may be shared
+      if $name == $group and !defined(Group[$group]) {
+        group { $group:
+          ensure     => 'absent',
+          forcelocal => $forcelocal,
+        }
       }
     }
+    accounts::key_management { "${name}_key_management":
+      ensure             => 'absent',
+      user               => $name,
+      user_home          => $_home,
+      sshkeys            => $sshkeys,
+      sshkey_custom_path => $sshkey_custom_path,
+      purge_user_home    => $purge_user_home,
+    }
   } else {
-    $_shell = $shell
-  }
-
-  # Check if user wants to create the group
-  if $create_group {
-    # Ensure that the group hasn't already been defined
-    if $ensure == 'present' and ! defined(Group[$group]) {
-      group { $group:
-        ensure     => $ensure,
-        gid        => $gid,
-        system     => $system,
-        forcelocal => $forcelocal,
-      }
-    # Only remove the group if it is the same as user name as it may be shared
-    } elsif $ensure == 'absent' and $name == $group {
-      group { $group:
-        ensure     => $ensure,
-        forcelocal => $forcelocal,
+    # Check if user wants to create the group
+    if $create_group {
+      Group[$group] -> User[$name]
+      # Ensure that the group hasn't already been defined
+      if ! defined(Group[$group]) {
+        group { $group:
+          ensure     => $ensure,
```

Comment:
Since this will always create the group perhaps it should be made 'present'?

Reply:
Sure.  So mote it be.